src/OutsideClickHandler.jsx

```diff
@@ -47,9 +47,9 @@ export default class OutsideClickHandler extends React.Component {
     if (!disabled) this.addMouseDownEventListener(useCapture);
   }
 
-  componentWillReceiveProps({ disabled, useCapture }) {
-    const { disabled: prevDisabled } = this.props;
-    if (prevDisabled !== disabled) {
+  componentDidUpdate(prevProps) {
+    const { disabled, useCapture } = this.props;
+    if (prevProps.disabled !== disabled) {
       if (disabled) {
         this.removeEventListeners();
       } else {
```

Comment:
will this change cause the mousedown event listener to be added *after* the DOM is updated? ie, could this change cause a user click to be missed?

Reply:
It should not. The events are independent of React, so adding/removing them doesn't depend on the stage of the component's lifecycle. Additionally, as you can see in the code snippet below (the code that gets called by this function) it applies the listener to `document`, which is independent of the component, so it doesn't care what state the component is in. In other words, clicking outside (anywhere in the document) is totally independent of the component; it needs to change when props change because the props might change to enable/disable it. Whether that happens before or after the render is irrelevant. 

```
  addMouseDownEventListener(useCapture) {
    this.removeMouseDown = addEventListener(
      document,
      'mousedown',
      this.onMouseDown,
      { capture: useCapture },
    );
  }
```

Reply:
Thanks, that's convincing.